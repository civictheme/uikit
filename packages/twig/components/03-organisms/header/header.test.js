const template = 'components/03-organisms/header/header.twig';

describe('Header Component', () => {
  test('renders with all attributes provided', async () => {
    const c = await dom(template, {
      theme: 'dark',
      content_top1: 'Top content 1',
      content_top2: 'Top content 2',
      content_top3: 'Top content 3',
      content_middle1: 'Middle content 1',
      content_middle2: 'Middle content 2',
      content_middle3: 'Middle content 3',
      content_bottom1: 'Bottom content 1',
      modifier_class: 'additional-class',
    });

    expect(c.querySelectorAll('.ct-header')).toHaveLength(1);
    expect(c.querySelector('.ct-header').classList).toContain('ct-theme-dark');
    expect(c.querySelector('.ct-header').classList).toContain('additional-class');
    expect(c.querySelector('.ct-header__content-top1').textContent).toContain('Top content 1');
    expect(c.querySelector('.ct-header__content-top2').textContent).toContain('Top content 2');
    expect(c.querySelector('.ct-header__content-top3').textContent).toContain('Top content 3');
    expect(c.querySelector('.ct-header__content-middle1').textContent).toContain('Middle content 1');
    expect(c.querySelector('.ct-header__content-middle2').textContent).toContain('Middle content 2');
    expect(c.querySelector('.ct-header__content-middle3').textContent).toContain('Middle content 3');
    expect(c.querySelector('.ct-header__content-bottom1').textContent).toContain('Bottom content 1');

    assertUniqueCssClasses(c);
  });

  test('renders with some attributes missing', async () => {
    const c = await dom(template, {
      theme: 'light',
      content_top1: 'Top content 1',
      content_top2: '',
      content_top3: '',
      content_middle1: '',
      content_middle2: 'Middle content 2',
      content_middle3: '',
      content_bottom1: 'Bottom content 1',
      modifier_class: '',
    });

    expect(c.querySelectorAll('.ct-header')).toHaveLength(1);
    expect(c.querySelector('.ct-header').classList).toContain('ct-theme-light');
    expect(c.querySelector('.ct-header__content-top1').textContent).toContain('Top content 1');
    expect(c.querySelector('.ct-header__content-top2')).toBeNull();
    expect(c.querySelector('.ct-header__content-top3')).toBeNull();
    expect(c.querySelector('.ct-header__content-middle1')).toBeNull();
    expect(c.querySelector('.ct-header__content-middle2').textContent).toContain('Middle content 2');
    expect(c.querySelector('.ct-header__content-middle3')).toBeNull();
    expect(c.querySelector('.ct-header__content-bottom1').textContent).toContain('Bottom content 1');

    assertUniqueCssClasses(c);
  });

  test('does not render when all slots are empty', async () => {
    const c = await dom(template, {
      theme: 'light',
      content_top1: '',
      content_top2: '',
      content_top3: '',
      content_middle1: '',
      content_middle2: '',
      content_middle3: '',
      content_bottom1: '',
      modifier_class: '',
    });

    expect(c.querySelectorAll('.ct-header')).toHaveLength(0);
  });
});
