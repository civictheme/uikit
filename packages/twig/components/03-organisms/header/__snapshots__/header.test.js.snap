// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component does not render when all slots are empty 1`] = `
<div>
  



</div>
`;

exports[`Header Component renders with all attributes provided 1`] = `
<div>
  


  
  <header
    class="ct-header ct-theme-dark additional-class"
    role="banner"
  >
    

          
    <div
      class="ct-header__top hide-xxs show-m"
    >
      

                              
      <div
        class="container"
      >
        
              
        <div
          class="row"
        >
          
                
          <div
            class="col-xxs-12"
          >
            
                  
            <div
              class="ct-header__content-top1"
            >
              
                    Top content 1
                  
            </div>
            
                
          </div>
          
              
        </div>
        
            
      </div>
      
                  
                  
      <div
        class="container"
      >
        
            
        <div
          class="row"
        >
          

                              
          <div
            class="col-xxs-12 col-m-4"
          >
            
                  
            <div
              class="ct-header__content-top2"
            >
              
                    Top content 2
                  
            </div>
            
                
          </div>
          
              
                              
          <div
            class="col-xxs-12 col-m-8"
          >
            
                  
            <div
              class="ct-header__content-top3"
            >
              
                    Top content 3
                  
            </div>
            
                
          </div>
          
              
            
        </div>
        
          
      </div>
      
              
    </div>
    
    
          
    <div
      class="ct-header__middle"
    >
      

                              
      <div
        class="container"
      >
        
              
        <div
          class="row"
        >
          
                
          <div
            class="col-xxs-12"
          >
            
                  
            <div
              class="ct-header__content-middle1"
            >
              
                    Middle content 1
                  
            </div>
            
                
          </div>
          
              
        </div>
        
            
      </div>
      
                  
                  
      <div
        class="container"
      >
        
            
        <div
          class="row row--no-wrap"
        >
          
                                                
          <div
            class="col col--no-grow"
          >
            
                    
            <div
              class="ct-header__content-middle2"
            >
              
                      Middle content 2
                    
            </div>
            
                  
          </div>
          
                              
                                                
          <div
            class="col"
          >
            
                    
            <div
              class="ct-header__content-middle3"
            >
              
                      Middle content 3
                    
            </div>
            
                  
          </div>
          
                                          
        </div>
        
          
      </div>
      
              
    </div>
    
    
                  
    <div
      class="container"
    >
      
          
      <div
        class="row"
      >
        
            
        <div
          class="col-xxs-12"
        >
          
              
          <div
            class="ct-header__content-bottom1"
          >
            
                Bottom content 1
              
          </div>
          
            
        </div>
        
          
      </div>
      
        
    </div>
    
          
  
  </header>
  

</div>
`;

exports[`Header Component renders with some attributes missing 1`] = `
<div>
  


  
  <header
    class="ct-header ct-theme-light "
    role="banner"
  >
    

          
    <div
      class="ct-header__top hide-xxs show-m"
    >
      

                              
      <div
        class="container"
      >
        
              
        <div
          class="row"
        >
          
                
          <div
            class="col-xxs-12"
          >
            
                  
            <div
              class="ct-header__content-top1"
            >
              
                    Top content 1
                  
            </div>
            
                
          </div>
          
              
        </div>
        
            
      </div>
      
                  
              
    </div>
    
    
          
    <div
      class="ct-header__middle"
    >
      

        
                  
      <div
        class="container"
      >
        
            
        <div
          class="row row--no-wrap"
        >
          
                                                
          <div
            class="col col--no-grow"
          >
            
                    
            <div
              class="ct-header__content-middle2"
            >
              
                      Middle content 2
                    
            </div>
            
                  
          </div>
          
                              
                          
        </div>
        
          
      </div>
      
              
    </div>
    
    
                  
    <div
      class="container"
    >
      
          
      <div
        class="row"
      >
        
            
        <div
          class="col-xxs-12"
        >
          
              
          <div
            class="ct-header__content-bottom1"
          >
            
                Bottom content 1
              
          </div>
          
            
        </div>
        
          
      </div>
      
        
    </div>
    
          
  
  </header>
  

</div>
`;
